server/sttp-stub-server/src/main/scala/sttp/tapir/server/stub/SttpStubServer.scala

```diff
@@ -0,0 +1,63 @@
+package sttp.tapir.server.stub
+
+import sttp.client.testing.SttpBackendStub
+import sttp.model.StatusCode
+import sttp.tapir.internal.SeqToParams
+import sttp.tapir.server.internal.{DecodeInputs, DecodeInputsResult, InputValues}
+import sttp.tapir.{DecodeFailure, Endpoint}
+
+trait SttpStubServer {
+
+  implicit class RichSttpBackendStub[F[_], S](val stub: SttpBackendStub[F, S]) {
+    def whenRequestMatches[E, O](endpoint: Endpoint[_, E, O, _]): TypeAwareWhenRequest[E, O] = {
+      new TypeAwareWhenRequest(
+        new stub.WhenRequest(req =>
+          DecodeInputs(endpoint.input, new SttpDecodeInputs(req)) match {
+            case DecodeInputsResult.Failure(_, _) => false
+            case DecodeInputsResult.Values(_, _)  => true
+          }
+        )
+      )
+    }
+
+    def whenInputMatches[I, E, O](endpoint: Endpoint[I, E, O, _])(inputMatcher: I => Boolean): TypeAwareWhenRequest[E, O] = {
+      new TypeAwareWhenRequest(
+        new stub.WhenRequest(req =>
+          DecodeInputs(endpoint.input, new SttpDecodeInputs(req)) match {
+            case DecodeInputsResult.Failure(_, _)  => false
+            case values: DecodeInputsResult.Values => inputMatcher(SeqToParams(InputValues(endpoint.input, values)).asInstanceOf[I])
+          }
+        )
+      )
+    }
+
+    def whenDecodingInputFailureMatches[E, O](
+        endpoint: Endpoint[_, E, O, _]
+    )(failureMatcher: PartialFunction[DecodeFailure, Boolean]): TypeAwareWhenRequest[E, O] = {
+      new TypeAwareWhenRequest(
+        new stub.WhenRequest(req => {
+          val result = DecodeInputs(endpoint.input, new SttpDecodeInputs(req))
+          result match {
+            case DecodeInputsResult.Failure(_, f) if failureMatcher.isDefinedAt(f) => failureMatcher(f)
+            case DecodeInputsResult.Values(_, _)                                   => false
+          }
+        })
+      )
+    }
+
+    def whenDecodingInputFailure[E, O](endpoint: Endpoint[_, E, O, _]): TypeAwareWhenRequest[E, O] = {
+      whenDecodingInputFailureMatches(endpoint) { case _ => true }
+    }
+
+    class TypeAwareWhenRequest[E, O](whenRequest: stub.WhenRequest) {
+
+      def thenSuccess(response: O): SttpBackendStub[F, S] =
```

Comment:
`thenRespond` and `thenRespondError` would be more in-line with current `SttpBackendStub`

Reply:
I know, but I wanted to give the feeling that this is well-typed stub and to distinguish it from simple `stub.WhenRequest`

Reply:
ok